packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPropertiesTests.m

```diff
@@ -19,7 +19,7 @@ - (void)testFLTGetFLTFlashModeForString {
   XCTAssertEqual(FLTFlashModeAuto, FLTGetFLTFlashModeForString(@"auto"));
   XCTAssertEqual(FLTFlashModeAlways, FLTGetFLTFlashModeForString(@"always"));
   XCTAssertEqual(FLTFlashModeTorch, FLTGetFLTFlashModeForString(@"torch"));
-  XCTAssertThrows(FLTGetFLTFlashModeForString(@"unkwown"));
+  XCTAssertEqual(FLTFlashModeInvalid, FLTGetFLTFlashModeForString(@"unknown"));
```

Comment:
Nit: should it be `FLTFlashModeUnknown`? in case we have new flash mode in the future.

Reply:
If we have a new flash mode in the future then we need to update both the Dart and Obj-C code at the same time. This code is purely for manual platform channel conversions, which means the only way to get a value we don't recognize is if someone adds [serialization for that mode on the Dart side](https://github.com/flutter/packages/blob/d0e9a0e1b3ac1f8afeba2a30da6d30fa06d96a48/packages/camera/camera_avfoundation/lib/src/avfoundation_camera.dart#L531) but not the corresponding deserialization code on the native side, which would make no sense.

If we have flash modes that the Dart side of `camera_avfoundation` has been updated to send over the method channel but the Obj-C side doesn't know about, then someone has made a serious mistake in adding a new flash mode. We shouldn't explicitly allow for that case with an "unknown", implying that's an okay state to be in.

Reply:
(This is why Pigeon will moot all of this; in Pigeon it's impossible for an enum value defined in the interface to exist on one side but not the other.)